src/components/screens/Consul/Proposals/NewProposal.js

```diff
@@ -0,0 +1,361 @@
+      summary: data?.summary,
+      videoUrl: data?.videoUrl
+    }
+  });
+
+  // GraphQL
+  const [submitProposal] = useMutation(START_PROPOSAL, {
+    client: ConsulClient
+  });
+  const [updateProposal] = useMutation(UPDATE_PROPOSAL, {
+    client: ConsulClient
+  });
+
+  useEffect(() => {
+    if (data?.tagList) {
+      kategorien.map((item) => {
+        for (let i = 0; i < data?.tagList.length; i++) {
+          const element = data?.tagList[i];
+          if (item.name === element) {
+            item.selected = true;
+          }
+        }
+      });
+      setTags(kategorien);
+    }
+  }, []);
+
+  useEffect(() => {
+    setTags(kategorien);
+    const selectedTag = tags.map((item) => {
+      if (item.selected) return item.name;
+    });
+    const filterData = selectedTag.filter((data) => data);
+    setValue('tagList', filterData.toString());
+  }, [tags]);
+
+  const onSubmit = async (val) => {
+    let variables = {
+      id: data?.id,
+      attributes: {
+        translationsAttributes: {
+          title: val.title,
+          summary: val.summary,
+          description: val.description
+        },
+        tagList: val.tagList,
+        termsOfService: termsOfService,
+        videoUrl: val.videoUrl
+      }
+    };
+
+    if (!termsOfService) return showRegistrationFailAlert();
+
+    switch (query) {
+      case queryTypes.START_PROPOSAL:
+        setStartLoading(true);
+        await submitProposal({
+          variables: variables
+        })
+          .then((val) => {
+            setStartLoading(false);
+            navigation.navigate(ScreenName.ConsulDetailScreen, {
+              query: QUERY_TYPES.CONSUL.PROPOSAL,
+              queryVariables: { id: val.data.submitProposal.id },
+              title: val.data.submitProposal.title,
+              publishedProposal: false
+            });
+          })
+          .catch((err) => {
+            graphqlErr(err.message);
+            console.error(err.message);
+            setStartLoading(false);
+          });
+        break;
+      case queryTypes.UPDATE_PROPOSAL:
+        setStartLoading(true);
+
+        //TODO: Mutation Error!
+        await updateProposal({
+          variables: variables
+        })
+          .then((val) => {
+            setStartLoading(false);
+            navigation.navigate(ScreenName.ConsulDetailScreen, {
+              query: QUERY_TYPES.CONSUL.PROPOSAL,
+              queryVariables: { id: val.data.updateProposal.id }
+            });
+          })
+          .catch((err) => {
+            graphqlErr(err.message);
+            console.error(err.message);
+            setStartLoading(false);
+          });
+        break;
+      default:
+        break;
+    }
+  };
+
+  if (startLoading) return <LoadingSpinner loading />;
+
+  return (
+    <SafeAreaViewFlex>
+      <WrapperWithOrientation>
+        {Inputs.map((item, index) => (
+          <Wrapper key={index} style={styles.noPaddingTop}>
+            {item.type === 'input' && (
```

Comment:
these type strings here could be constants at some place and used here to avoid hardcoding these strings on maybe multiple places.

Reply:
Are you talking about creating a new helper?

Reply:
Yes you are right. In this way, I would have built a better structure. I am updating the code.

Reply:
2ef6703408d8cec794591fcfff91d2b506152d3b

Reply:
yes